content/concepts/consensus-network/transaction-malleability.md

```diff
@@ -0,0 +1,142 @@
+
+    Three ways that a transaction may not enable the tfFullyCanonicalSig flag are:
+
+    - The system explicitly specifies a `Flags` field that does not have the tfFullyCanonicalSig bit enabled.
+    - The transaction is multi-signed and does not explicitly enable the tfFullyCanonicalSig flag.
+    - The system omits the `Flags` field from the transaction fields, but uses a non-standard implementation that does not automatically enable tfFullyCanonicalSig when signing.
+
+    To be vulnerable, the transaction must be signed with an ECDSA key pair. If multi-signed, the transaction must be signed by at least one ECDSA key pair.
+
+    Most likely, the vulnerable transaction uses a fully-validated signature, but the flags indicate that the transaction would also be valid with a non-fully-canonical one. The transaction may also use `LastLedgerSequence` so that its final outcome is clear in a finite amount of time.
+
+2. The system notes the identifying hash of the vulnerable transaction, submits it to the XRP Ledger network, then begins monitoring for that hash to be included in a validated ledger version.
+
+3. A malicious actor sees the transaction propagating through the network before it becomes confirmed.
+
+4. The malicious actor calculates the alternate signature for the vulnerable transaction, and derives the hash of the resulting transaction data.
+
+    Unlike creating a signature for a different transaction data, this does not require a large amount of computational work. It can be done in much less time than it takes to generate a signature in the first place.
+
+5. The malicious actor submits the modified (likely non-fully-canonical) transaction to the network.
+
+    This creates a "race" between the transaction as originally submitted and the modified version submitted by the malicious actor. The two transactions are mutually exclusive. Both are valid, but they have the same exact transaction data, including the `Sequence` number, so at most one of them can ever be included in a validated ledger.
+
+    Servers in the peer-to-peer network have no way of knowing which one "came first" or was intended by its original sender. Delays or other coincidences in network connectivity could result in validators seeing only one or the other by the time they finalize their consensus proposals, so either one could "win the race".
+
+    A malicious actor could increase the chances of getting non-canonical transactions confirmed if they controlled some number of well-connected servers in the peer-to-peer network, even if those servers are not trusted as validators.
+
+    If the malicious actor controls the only server to which the vulnerable system submitted the transaction, the malicious actor can easily control which version is distributed to the rest of the network.
+
+6. The malicious actor's version of the transaction achieves a consensus and becomes included in a validated ledger.
```

Comment:
Do you want the "a" between "achieves" and "consensus"?

Reply:
_shrug_

I guess it works just as well without, so may as well remove it.